src/dscanner/analysis/lambda_return_check.d

```diff
@@ -32,7 +31,31 @@ extern (C++) class LambdaReturnCheck(AST) : BaseAnalyzerDmd
 		import dmd.globals : global;
 		import dmd.lexer : Lexer;
 
-		auto bytes = readFile(fileName) ~ '\0';
+		auto rawCode = readFile(fileName) ~ '\0';
+		ubyte[] bytes = rawCode;
+
+		version (Windows)
```

Comment:
As I stated on the N.G. this change is likely motivated for the wrong reason.

At the very least it should be split out.

``std.file : write`` does not introduce new lines as described in your post.
https://github.com/dlang/phobos/blob/master/std/file.d#L861
https://forum.dlang.org/post/dijbuujbcfdszsxqpuor@forum.dlang.org

Reply:
It looks like Windows API `WriteFile` prepends a `\r` char before every `\n` char, even if a carriage return already exists. We have switched from using `write` to using `rawWrite` in our unit tests:
https://github.com/Dlang-UPB/D-scanner/commit/8db5a658134994cd1f388ca4b5613a5bb24870a2

Reply:
``WriteFile`` shouldn't be doing that, it's for raw byte data.

It does not know that the input is text, and if it did that it would corrupt the output.

Reply:
>`WriteFile` shouldn't be doing that
But is there any way to actually confirm or infirm this? I don't see anything specified in the public docs:
https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-writefile

Reply:
![Capture](https://github.com/user-attachments/assets/4d177f2b-54ac-4fa5-b4ee-5c7dd32d5a71)

```d
import core.sys.windows.windows;

void main() {
	HANDLE h = CreateFileW("file.txt"w.ptr, GENERIC_READ | GENERIC_WRITE, 0, null, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, null);
	WriteFile(h, "Hello\nWorld!".ptr, 12, null, null);
	CloseHandle(h);
}
```